### ISSUE TITLE
Picking the right similarity_space_type name to create an inner product index

### ISSUE THREAD

**Issue by brandenchan** [opendistro-for-elasticsearch/k-NN#350]
Hi ,
 
I am trying to create an ANN index that uses inner product as the similarity function. As far as I can see, the support for this has been implemented (#144). However, I have not found a value for `knn.space_type` that is accepted. The documentation [here](https://opendistro.github.io/for-elasticsearch-docs/docs/knn/approximate-knn/#spaces) doesn't seem to list inner product.

For whatever value of `knn.space_type` I give, I am constantly getting 
```
RequestError(400, 'invalid_parameter_exception', 'Unsupported space type: inner_product')
```

Any help would be greatly appreciated!

**Comment by jmazanec15** [comment]
Hi @brandenchan 

Inner product is not available in ODFE 1.13, which was our last ODFE release: https://github.com/opendistro-for-elasticsearch/k-NN/blob/opendistro-1.13/src/main/java/com/amazon/opendistroforelasticsearch/knn/index/SpaceTypes.java. We added it to main branch after ODFE 1.13, but then with the OpenSearch fork, it was decided there would be no more ODFE releases.

It is available in OpenSearch rc1 and will be available in OpenSearch GA. Sorry for the confusion.

**Comment by brandenchan** [comment]
Great thanks!
